src/cdk-experimental/scrolling/virtual-scroll-viewport.ts

```diff
@@ -137,6 +137,15 @@ export class CdkVirtualScrollViewport implements DoCheck, OnInit, OnDestroy {
     this._detachedSubject.complete();
   }
 
+  /** Update the viewport dimensions and re-render. */
+  updateViewport() {
+    const viewportEl = this.elementRef.nativeElement;
+    this._viewportSize = this.orientation === 'horizontal' ?
```

Comment:
since this also appears in the Promise above, can we extract it into a separate `_measureViewportSize` method